src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/DeleteToolbarAction.js

```diff
@@ -55,45 +67,124 @@ export default class DeleteToolbarAction extends AbstractFormToolbarAction {
+                requestOptions={this.deleteDependantsDialogRequestOptions}
+            />
+        );
+    }
+
+    handleDialogCancel = () => {
+        this.closeDialog();
+    };
+
+    handleDialogConfirm = () => {
+        this.delete();
+    };
+
+    @action closeDialog = () => {
+        this.showDialog = false;
+    };
+
+    renderDialog(postfix: string) {
+        return (
+            <Dialog
+                cancelText={translate('sulu_admin.cancel')}
+                confirmLoading={this.resourceFormStore.deleting}
+                confirmText={translate('sulu_admin.ok')}
+                onCancel={this.handleDialogCancel}
+                onConfirm={this.handleDialogConfirm}
+                open={this.showDialog}
+                title={translate('sulu_admin.delete' + postfix + '_warning_title')}
+            >
+                {translate('sulu_admin.delete' + postfix + '_warning_text')}
+            </Dialog>
+        );
+    }
+
     getNode() {
         const {delete_locale: deleteLocale = false} = this.options;
         const postfix = deleteLocale ? '_locale' : '';
 
         return (
             <Fragment key={'sulu_admin.delete' + postfix}>
-                <Dialog
-                    cancelText={translate('sulu_admin.cancel')}
-                    confirmLoading={this.resourceFormStore.deleting}
-                    confirmText={translate('sulu_admin.ok')}
-                    onCancel={this.handleCancel}
-                    onConfirm={this.handleConfirm}
-                    open={this.showDialog}
-                    title={translate('sulu_admin.delete' + postfix + '_warning_title')}
-                >
-                    {translate('sulu_admin.delete' + postfix + '_warning_text')}
-                </Dialog>
-                <Dialog
-                    cancelText={translate('sulu_admin.cancel')}
-                    confirmLoading={this.resourceFormStore.deleting}
-                    confirmText={translate('sulu_admin.ok')}
-                    onCancel={this.allowConflictDeletion ? this.handleLinkCancel : undefined}
-                    onConfirm={this.handleLinkConfirm}
-                    open={this.showLinkedDialog}
-                    title={this.allowConflictDeletion
-                        ? translate('sulu_admin.delete_linked_warning_title')
-                        : translate('sulu_admin.item_not_deletable')
-                    }
-                >
-                    {this.allowConflictDeletion
-                        ? translate('sulu_admin.delete_linked_warning_text')
-                        : translate('sulu_admin.delete_linked_abort_text')
-                    }
-                    <ul>
-                        {this.referencingItems.map((referencingItem, index) => (
-                            <li key={index}>{referencingItem.name}</li>
-                        ))}
-                    </ul>
-                </Dialog>
+                {this.renderDialog(postfix)}
```

Comment:
could we extract these three dialogs (normal delete, delete referenced and delete dependants) into a component that can be used in the `DeleteToolbarAction` and also the `List`?

Reply:
I'm not sure if I would do that, because the logic for deletion is quite different ... The only thing that's really the same is the `sulu_admin.ok` and the `sulu_admin.cancel` translations for the buttons